pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -408,21 +436,54 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 
 func updateTaskRunsStatus(pr *v1alpha1.PipelineRun, pipelineState []*resources.ResolvedPipelineRunTask) {
 	for _, rprt := range pipelineState {
+		if rprt.TaskRun == nil && rprt.ResolvedConditionChecks == nil {
+			continue
+		}
+		var prtrs *v1alpha1.PipelineRunTaskRunStatus
 		if rprt.TaskRun != nil {
-			prtrs := pr.Status.TaskRuns[rprt.TaskRun.Name]
-			if prtrs == nil {
-				prtrs = &v1alpha1.PipelineRunTaskRunStatus{
-					PipelineTaskName: rprt.PipelineTask.Name,
-				}
-				pr.Status.TaskRuns[rprt.TaskRun.Name] = prtrs
+			prtrs = pr.Status.TaskRuns[rprt.TaskRun.Name]
+		}
+		if prtrs == nil {
+			prtrs = &v1alpha1.PipelineRunTaskRunStatus{
+				PipelineTaskName: rprt.PipelineTask.Name,
 			}
+		}
+
+		if rprt.TaskRun != nil {
 			prtrs.Status = &rprt.TaskRun.Status
 		}
+
+		if len(rprt.ResolvedConditionChecks) > 0 {
+			cStatus := make(map[string]*v1alpha1.PipelineRunConditionCheckStatus)
+			for _, c := range rprt.ResolvedConditionChecks {
+				cStatus[c.ConditionCheckName] = &v1alpha1.PipelineRunConditionCheckStatus{
+					ConditionName: c.Condition.Name,
+				}
+				if c.ConditionCheck != nil {
+					ccStatus := v1alpha1.ConditionCheckStatus(c.ConditionCheck.Status)
+					cStatus[c.ConditionCheckName].Status = &ccStatus
+				}
+			}
+			prtrs.ConditionChecks = cStatus
+			if rprt.ResolvedConditionChecks.IsComplete() && !rprt.ResolvedConditionChecks.IsSuccess() {
+				if prtrs.Status == nil {
+					prtrs.Status = &v1alpha1.TaskRunStatus{}
+				}
+				prtrs.Status.SetCondition(&apis.Condition{
+					Type:    apis.ConditionSucceeded,
+					Status:  corev1.ConditionFalse,
+					Reason:  resources.ReasonConditionCheckFailed,
+					Message: fmt.Sprintf("ConditionChecks failed for Task %s in PipelineRun %s", rprt.TaskRunName, pr.Name),
+				})
+			}
+		}
+		pr.Status.TaskRuns[rprt.TaskRunName] = prtrs
 	}
 }
 
 func (c *Reconciler) updateTaskRunsStatusDirectly(pr *v1alpha1.PipelineRun) error {
 	for taskRunName := range pr.Status.TaskRuns {
+		// TODO: Add conditionCheck statuses here
```

Comment:
Is this a TODO for these commits or intended to tackle later?  If to tackle later, suggest updating with `TODO(dibyom)` to guide future developers on who to contact for more info.